utils/src/main/scala/za/co/absa/enceladus/utils/validation/SchemaPathValidator.scala

```diff
@@ -141,54 +141,51 @@ object SchemaPathValidator {
                                       parentOnly: Boolean = false,
                                       fullPathNew: Boolean = false,
                                       parentPath: String = ""): Seq[ValidationIssue] = {
-    // scalastyle:off  return Covered in Issue #730
     if (path.isEmpty) {
-      return Nil
+      Nil
     }
+    else {
+
+      val currentField = path(0)
+      val fullPath = s"$parentPath${path.mkString(".")}"
+      if (parentOnly && path.length == 1) {
+        schema.fields match {
+          case f if fullPathNew && f.exists(field => field.name == currentField) =>
```

Comment:
I think it would be more effective to go through the fields just once and only in case of insensitive hit check witch message to send based on sensitive match or not.
(We have datasets with thousands of columns)